### PR TITLE
lib/libc/math: Add lrint, lrintf and lrintl API implementation to libc

### PR DESCRIPTION
**Pull request by neel-samsung**
Add lrint, lrintf and lrintl API implementation to libc. These APIs shall round their argument to the nearest integer value, rounding according to the current rounding direction.

### CODE REVIEW

**Review by sunghan-chang — COMMENTED** [review]
1. Please fix violations in coding rules.
2. Could you add test result at commit description?
    1. How will you verify your implementation? Will make UTCs for them?

**Comment by neel-samsung** [comment]
> 1. Please fix violations in coding rules.
> 2. Could you add test result at commit description?
>    
>    1. How will you verify your implementation? Will make UTCs for them?

Thank you for Review.

1. I have fixed coding guideline violation. Please let me know if I have missed any.
2. I have also added Kernel TCs for these APIs Testing and added the result in commit description.

**Comment by sunghan-chang** [comment]
@neel-samsung You should not mix different topics in a single commit. Please split adding libc and adding tc for them.

File: lib/libc/math/lib_lrint.c [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+ ************************************************************************/
+
+#ifdef CONFIG_HAVE_DOUBLE
+long int lrint(double x) 
+{
+	long int ret;
+	double intpart, fracpart;
+
+	/* If x is NaN, infinite, -infinite, too large or too small, domain error is returned */
+
```
Comment (edwakuwaku) [inline lib/libc/math/lib_lrint.c:38]: It seems added many unnecessary spaces

File: lib/libc/math/lib_lrint.c [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	if (isnan(x) || isinf(x) || x > (double) LONG_MAX || x < (double) LONG_MIN) {
```
Comment (sunghan-chang) [inline lib/libc/math/lib_lrint.c:39]: Almost type casting has a space. Let's remove it from all.

**Comment by sunghan-chang** [comment]
@neel-samsung Could you check comment to merge by today?

File: lib/libc/math/lib_lrintf.c [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+#include <tinyara/compiler.h>
+
+/************************************************************************
+ * Public Functions
+ ************************************************************************/
+
+long int lrintf(float x) 
+{
+	long int ret;
+	float intpart, fracpart;
```
Comment (edwakuwaku) [inline lib/libc/math/lib_lrintf.c:34]: similar here
float intpart;
float fracpart;

**Comment by neel-samsung** [comment]
> @neel-samsung Could you check comment to merge by today?

Okay, I will update it by today.

File: lib/libc/math/lib_lrintf.c [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+	 * positive infinity, rintf() is identical to ceilf()
+	 * with type conversion to long.
+	 */
+#if defined(CONFIG_FP_ROUND_POSITIVE) && CONFIG_FP_ROUNDING_POSITIVE != 0
+	ret = (long int)ceilf(x);
+#elif defined(CONFIG_FP_ROUND_NEGATIVE) && CONFIG_FP_ROUNDING_NEGATIVE != 0
+	ret = (long int)floorf(x);
+#else
+	fracpart = modff(x, &intpart);	/* Separate the integer and fractional parts */
+	if(x >= 0.0) {
```
Comment (edwakuwaku) [inline lib/libc/math/lib_lrintf.c:55]: if (x >= 0.0)

File: lib/libc/math/lib_lrintl.c [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+	 * positive infinity, rintl() is identical to ceill()
+	 * with type conversion to long.
+	 */
+#if defined(CONFIG_FP_ROUND_POSITIVE) && CONFIG_FP_ROUNDING_POSITIVE != 0
+	ret = (long int)ceill(x);
+#elif defined(CONFIG_FP_ROUND_NEGATIVE) && CONFIG_FP_ROUNDING_NEGATIVE != 0
+	ret = (long int)floorl(x);
+#else
+	fracpart = modfl(x, &intpart);	/* Separate the integer and fractional parts */
+	if(x >= 0.0) {
```
Comment (edwakuwaku) [inline lib/libc/math/lib_lrintl.c:56]: if (x >= 0.0)

File: lib/libc/math/lib_lrint.c [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+#include <math.h>
+#include <errno.h>
+#include <tinyara/compiler.h>
+
+/************************************************************************
+ * Public Functions
+ ************************************************************************/
+
+#ifdef CONFIG_HAVE_DOUBLE
```
Comment (kishore-sn) [inline lib/libc/math/lib_lrint.c:31]: Let us add #include <tinyara/config.h>as the first include file always to get the config values.

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**